### ISSUE TITLE
Error compiling this repo

### ISSUE THREAD

**Issue by MikeMannox** [tekartik/sqflite#24]
Hi, 

Thanks for your work on this project.

I was trying to use your project, however I keep getting an error.
To try and troubleshoot I cloned your project, and found I get the same issue.

I am using a Mac, and the steps I took:

1. Upgraded `flutter-sdk`
2. Ran `flutter doctor`
3. cloned your repo
4. open project in Android Studio 
5. ran `flutter packages get`
6. attempted a compile (using Android Emulator)

The console error message below:

```bash
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...
Resolving dependencies...
* Error running Gradle:
Exit code 1 from: /Users/Home/Flutter/sqflite/example/android/gradlew app:properties:
Parallel execution is an incubating feature.

BUILD FAILED

Total time: 1.298 secs


FAILURE: Build failed with an exception.

* Where:
Build file '/Users/Home/Flutter/sqflite/example/android/build.gradle' line: 3

* What went wrong:
A problem occurred evaluating root project 'android'.
> Could not find method google() for arguments [] on repository container.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

Finished with error: Please review your Gradle project setup in the android/ folder.
```

**Comment by MikeMannox** [comment]
note:
In my own project I did bump versions to try:
`classpath 'com.android.tools.build:gradle:3.0.1'`
`distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip`
`compileSdkVersion 26`
`buildToolsVersion '26.0.3'`
and then get another error:
```bash
FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Failed to transform file 'flutter-x86.jar' to match attributes {artifactType=android-classes} using transform JarTransform
   > Transform output file /Users/Home/Flutter/my_app/build/app/intermediates/flutter/flutter-x86.jar does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 9s
Finished with error: Gradle build failed: 1
```

**Comment by alextekartik** [comment]
Sorry for the trouble. Flutter build system is not perfect yet. Indeed the error you're getting is about using an old gradle version. I'm personally using the gradle wrapper that i have not checked in to mimic what google does with in their flutter plugin. I remember the gradle upgrade (to gradle 4.1) process was painful for me too. Hard for me to remember where and how help you. I only build on Ubuntu and MacOS. Maybe make sure you are not using a local gradle distribution (`gradle --version` or `./gradlew --version`) and it you are, upgrade it. gradle 2.x version won't support creating a 4.x wrapper version.

Another solution is to create an android projet and copy the gradle generated files `gradlew`, `gradlew.bat` and `gradlew.wrapper/` into `sqlite/example/android`.

I think I use the same gradle than google [flutter plugins](https://github.com/flutter/plugins). You might want to try to see if their example project build properly and if yes, copy the gradle generated files

**Comment by MikeMannox** [comment]
@alextekartik - Thanks for the reply.

I have managed to compile the project this morning. I changed
`distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip`
in  
`example/android/gradle.wrapper/gradle-wrapper.properties`
cleaned the project and compiled and it worked.

Now I will need to work backwards and try to figure out which of my imported dependencies is making a problem for me in my own project.

Thank you.
